docs/guide/connecting.asciidoc

```diff
@@ -3,58 +3,155 @@
+
+When you start Elasticsearch for the first time you'll see a distinct block like the one below in the output from Elasticsearch (you may have to scroll up if it's been a while):
+
+```
+----------------------------------------------------------------------
+-> Elasticsearch security features have been automatically configured!
+-> Authentication is enabled and cluster connections are encrypted.
+
+->  Password for the elastic user (reset with `bin/elasticsearch-reset-password -u elastic`):
+  lhQpLELkjkrawaBoaz0Q
+
+->  HTTP CA certificate SHA-256 fingerprint:
+  a52dd93511e8c6045e21f16654b77c9ee0f34aea26d9f40320b531c474676228
+...
+----------------------------------------------------------------------
+```
+
+Note down the `elastic` user password and HTTP CA fingerprint for the next sections. In the examples below they will be stored in the variables `ELASTIC_PASSWORD` and `CERT_FINGERPRINT` respectively.
+
+Depending on the circumstances there are two options for verifying the HTTPS connection, either verifying with the CA certificate itself or via the HTTP CA certificate fingerprint.
+
+[discrete]
+==== Verifying HTTPS with CA certificates
+
+Using the `ca_certs` option is the default way the Python Elasticsearch client verifies an HTTPS connection.
+
+The generated root CA certificate can be found in the `certs` directory in your Elasticsearch config location. If you're running Elasticsearch in Docker there is https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html[additional documentation for retrieving the CA certificate].
+
+Once you have the `http_ca.crt` file somewhere accessible pass the path to the client via `ca_certs`:
+
+```python
 from elasticsearch import Elasticsearch
 
-# Single node via URL
-es = Elasticsearch("http://localhost:9200")
+# Generated by Elasticsearch
+ELASTIC_PASSWORD = "<password>"
 
-# Multiple nodes via URL
-es = Elasticsearch([
-    "http://localhost:9200",
-    "http://localhost:9201",
-    "http://localhost:9202"
-])
+# Create the client instance
+client = Elasticsearch(
+  "https://localhost:9200",
+  ca_certs="/path/to/http_ca.crt",
+  basic_auth=("elastic", ELASTIC_PASSWORD)
+)
 
-# Single node via dictionary
-es = Elasticsearch({"scheme": "http", "host": "localhost", "port": 9200})
+# Successful response!
+client.info()
+# {'name': 'instance-0000000000', 'cluster_name': ...}
+```
 
-# Multiple nodes via dictionary
-es = Elasticsearch([
-    {"scheme": "http", "host": "localhost", "port": 9200},
-    {"scheme": "http", "host": "localhost", "port": 9201},
-])
-----------------------------
+NOTE: If you don't specify `ca_certs` or `ssl_assert_fingerprint` then the https://certifiio.readthedocs.io[certifi package] will be used for `ca_certs` by default if available.
 
 [discrete]
-[[connect-ec]]
-==== Connecting to Elastic Cloud
+==== Verifying HTTPS with certificate fingerprints
```

Comment:
Should we add `(Python 3.10 or later)` to make it even more obvious than the NOTE?